app-backend/db/src/main/scala/com/azavea/rf/database/SceneWithRelatedDao.scala

```diff
@@ -0,0 +1,216 @@
+    ThumbnailDao.query.filter(fr"scene = ${sceneId}").list
+
+  def sceneToSceneWithRelated(scene: Scene, queryParams: CombinedSceneQueryParams): ConnectionIO[Scene.WithRelated] = {
+    // "The astute among you will note that we don’t actually need a monad to do this;
+    // an applicative functor is all we need here."
+    // let's roll, doobie
+    val componentsIO: ConnectionIO[(List[Image.WithRelated], List[Thumbnail])] = {
+      val thumbnails = getSceneThumbnails(scene.id)
+      val images = getSceneImages(scene.id, queryParams.imageParams)
+      (images, thumbnails).tupled
+    }
+
+    componentsIO map {
+      case (images, thumbnails) => scene.withRelatedFromComponents(images, thumbnails)
+    }
+  }
+
+  def listScenes(pageRequest: PageRequest, sceneParams: CombinedSceneQueryParams, user: User): ConnectionIO[PaginatedResponse[Scene.WithRelated]] = {
+
+    val pageFragment: Fragment = Page(pageRequest)
+    val queryFilters: List[Option[Fragment]] = makeFilters(List(sceneParams)).flatten
+    println((selectF ++ Fragments.whereAndOpt(queryFilters: _*) ++ pageFragment).toString)
+    val scenesIO: ConnectionIO[List[Scene]] =
+      (selectF ++ Fragments.whereAndOpt(queryFilters: _*) ++ pageFragment)
+        .query[Scene]
+        .stream
+        .compile
+        .toList
+    val withRelatedsIO: ConnectionIO[List[Scene.WithRelated]] = scenesIO flatMap {
+      case scenes =>
+        scenes.traverse(sceneToSceneWithRelated(_, sceneParams))
+    }
+    val countIO: ConnectionIO[Int] =
+      (fr"SELECT count(*) FROM scenes" ++ Fragments.whereAndOpt(queryFilters: _*)).query[Int].unique
+
+    for {
+      _ <- println("getting scenes").pure[ConnectionIO]
+      page <- withRelatedsIO
+      _ <- println("counting").pure[ConnectionIO]
+      count <- countIO
+    } yield {
+      val hasPrevious = pageRequest.offset > 0
+      val hasNext = (pageRequest.offset * pageRequest.limit) + 1 < count
+      PaginatedResponse[Scene.WithRelated](count, hasPrevious, hasNext, pageRequest.offset, pageRequest.limit, page)
+    }
+  }
+
+  def getScene(sceneId: UUID, user: User): ConnectionIO[Option[Scene.WithRelated]] = {
+    val queryFilters = List(Some(fr"id = ${sceneId}"), ownerEditFilter(user))
+    listQuery(queryFilters, None).query[Scene.WithRelated].option
+  }
+
+  def getScenesToIngest(projectId: UUID): ConnectionIO[List[Scene.WithRelated]] = {
+    val fragments = List(
+      Some(fr"""(ingest_status = ${IngestStatus.ToBeIngested.toString} :: ingest_status
+           OR (ingest_status = ${IngestStatus.Ingesting.toString} :: ingest_status AND (now() - modified_at) > '1 day'::interval))
+        """),
+      Some(fr"scenes.id IN (SELECT scene_id FROM scenes_to_projects WHERE project_id = ${projectId})")
+    )
+    listQuery(fragments, None).query[Scene.WithRelated].list
+  }
+
+  def listQuery(fragments: List[Option[Fragment]], page: Option[PageRequest]): Fragment = {
```

Comment:
I'm fairly sure this can be removed entirely but haven't checked and, as with not throwing an error on conversion to filters, am trying to keep the changeset sort of focused. I'll try removing entirely after review

Reply:
...it definitely cannot and that was a foolish thought :tada: